.github/workflows/testpackage.yml

```diff
@@ -0,0 +1,21 @@
+name: Test Python Package
```

Comment:
I don't really like having the test execution defined twice.

Reply:
Me neither. I've refactored this to just be one workflow and the deploy bit to only execute if we're actually pushing a commit to master.

In general, there's a [desire](https://github.community/t/reusing-sharing-inheriting-steps-between-jobs-declarations/16851) in the community to have the ability to share steps between workflows without needing to author a full blown action or pack everything into one workflow, but this seems to not yet be supported.